Sources/NIO/EventLoopFuture.swift

```diff
@@ -374,20 +374,23 @@ public final class EventLoopFuture<Value> {
     @usableFromInline
     internal var _value: Result<Value, Error>? {
         didSet {
-            self._isFulfilled.store(true)
+            debugOnly {
+                self._isFulfilled!.store(true)
+            }
         }
     }
 
     @usableFromInline
-    internal let _isFulfilled: UnsafeEmbeddedAtomic<Bool>
+    internal let _isFulfilled: UnsafeEmbeddedAtomic<Bool>?
 
     /// The `EventLoop` which is tied to the `EventLoopFuture` and is used to notify all registered callbacks.
     public let eventLoop: EventLoop
 
     /// Whether this `EventLoopFuture` has been fulfilled. This is a thread-safe
     /// computed-property.
     internal var isFulfilled: Bool {
-        return self._isFulfilled.load()
+        precondition(_isDebugAssertConfiguration())
+        return self._isFulfilled!.load()
```

Comment:
One note here: this makes it impossible to run our unit tests in release mode. I don't care about this, but others might. We should document the fact that this operations crashes when run in release mode.

Reply:
ah shit, that's right. I'll remove this.

Reply:
killed this completely.